>From 8100c085cfe4118a2c5ddd3b6da70b30000b125f Mon Sep 17 00:00:00 2001
From: HAT <hat@fa2.so-net.ne.jp>
Date: Sat, 23 Jul 2016 01:14:28 +0900
Subject: [PATCH] afpd: cannot build when ldap is not defined

related commit
e9b0ebaed27091da6a8884fe8267be209bf4fbc0
afpd: don't use network IDs without LDAP

Bug: https://sourceforge.net/p/netatalk/bugs/630/

Signed-off-by: HAT <hat@fa2.so-net.ne.jp>
---
 NEWS              | 4 ++++
 etc/afpd/volume.c | 7 +++++++
 2 files changed, 11 insertions(+)

diff --git a/NEWS b/NEWS
index 1ef9631..9ccccdb 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,7 @@
+Changes in 3.1.10
+================
+* FIX: cannot build when ldap is not defined, bug #630
+
 Changes in 3.1.9
 ================
 * FIX: afpd: fix "admin group" option
diff --git a/etc/afpd/volume.c b/etc/afpd/volume.c
index 8665d80..cb9d123 100644
--- a/etc/afpd/volume.c
+++ b/etc/afpd/volume.c
@@ -40,7 +40,10 @@
 #include <atalk/unix.h>
 #include <atalk/netatalk_conf.h>
 #include <atalk/server_ipc.h>
+
+#ifdef HAVE_LDAP
 #include <atalk/ldapconfig.h>
+#endif /* HAVE_LDAP */
 
 #ifdef CNID_DB
 #include <atalk/cnid.h>
@@ -380,8 +383,12 @@ static int getvolparams(const AFPObj *obj, uint16_t bitmap, struct vol *vol, str
                         ashort |= VOLPBIT_ATTR_UNIXPRIV;
                     if (vol->v_flags & AFPVOL_TM)
                         ashort |= VOLPBIT_ATTR_TM;
+#ifdef HAVE_LDAP
                     if (!ldap_config_valid || vol->v_flags & AFPVOL_NONETIDS)
                         ashort |= VOLPBIT_ATTR_NONETIDS;
+#else
+                    ashort |= VOLPBIT_ATTR_NONETIDS;
+#endif
                     if (obj->afp_version >= 32) {
                         if (vol->v_vfs_ea)
                             ashort |= VOLPBIT_ATTR_EXT_ATTRS;
-- 
2.9.2

